src/main/java/frc/robot/subsystems/TurretS.java

```diff
@@ -0,0 +1,186 @@
+// Copyright (c) FIRST and other WPILib contributors.
+// Open Source Software; you can modify and/or share it under the terms of
+// the WPILib BSD license file in the root directory of this project.
+
+package frc.robot.subsystems;
+
+import com.revrobotics.CANSparkMax;
+import com.revrobotics.RelativeEncoder;
+import com.revrobotics.SparkMaxRelativeEncoder;
+import com.revrobotics.CANSparkMax.SoftLimitDirection;
+import com.revrobotics.CANSparkMaxLowLevel.MotorType;
+
+import edu.wpi.first.math.MathUtil;
+import edu.wpi.first.math.controller.PIDController;
+import edu.wpi.first.wpilibj.DigitalInput;
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.robot.Constants;
+import io.github.oblarg.oblog.Loggable;
+
+/**
+ * The turret subsystem. It contains methods to get encoder counts and converts
+ * them to degrees.
+ * 
+ * @author Benjamin Su
+ * @author Noah Kim
+ */
+public class TurretS extends SubsystemBase implements Loggable {
+  private CANSparkMax sparkMax = new CANSparkMax(Constants.CAN_ID_TURRET, MotorType.kBrushless);
+  private RelativeEncoder sparkMaxEncoder = sparkMax.getEncoder(SparkMaxRelativeEncoder.Type.kHallSensor, 42);
+  private DigitalInput limitSwitch = new DigitalInput(Constants.LIMIT_SWITCH_PORT_NUMBER);
+  private PIDController pid = new PIDController(0.0025, 0, 0);
+
+  /** Creates a new TurretS. */
```

Comment:
can this be expanded? maybe put a description and authors

Reply:
See the class javadoc.